src/sql-parser/src/parser.rs

```diff
@@ -40,6 +41,9 @@ use crate::lexer::{self, Token};
 // a healthy factor to be conservative.
 const RECURSION_LIMIT: usize = 128;
 
+/// Maximum allowed size for a batch of statements
+pub const MAX_STATEMENT_BATCH_SIZE: ByteSize = ByteSize::mb(1);
```

Comment:
Not convinced we need a crate here.

```suggestion
// 1MiB.
pub const MAX_STATEMENT_BATCH_SIZE: usize= 1 << 20;
```

Reply:
The one nice thing the crate provides is human readable formatting. So for example, the error message currently looks like this
> "statement batch size cannot exceed 1000.0 KB"

Without any formatting of our own. Though we can just always format the error message in terms of bytes and then we wouldn't need that.

Reply:
Ah, that's worth keeping then. Ok, ignore my code suggestions.

Reply:
Ehhh but it's also like so easy to do ourselves? Whichevery.

Reply:
I'll get rid of it. I was already on the fence because I didn't like all the casts.

Reply:
Maybe I'll remove it from the const and use it in the error message.